### DISCUSSION TITLE
How protect all pages with login shield

### DISCUSSION THREAD

**Discussion by caosdp-rs** [codeigniter4/shield#263] [Q&A] [answered]
I need to protect all pages to only be accessed after successful login, except the page (login, register)
I tried to add in Routes.php
```php
service('auth')->routes($routes, ['except' => ['login', 'register']]);
```

But it didn't work, do I need to change it somewhere else?

**Comment by kenjis** [comment, marked as answer]
Configure globals filter.
See https://codeigniter4.github.io/CodeIgniter4/incoming/filters.html#globals

**Reply by caosdp-rs** [reply]
'chain' => [
            'before' => [
                '/',
                'home(/*)?',
                'users(/*)?',
            ]
        ],

it's safe that way ???

**Reply by kenjis** [reply]
No. I recommend you access with your browser and confirm the pages are protected.

The meta characters that you can use is only `*`.

> to protect all pages to only be accessed after successful login, except the page (login, register)

```php
    public $globals = [
        'before' => [
            'session' => ['except' => ['login', 'register']],
        ],
        'after' => [],
    ];
```

**Reply by nenofetch** [reply]
how to redirect into another page after login? i was adding this to my filters but it goes return into login again

**Reply by datamweb** [reply]
> how to redirect into another page after login? 

See https://github.com/codeigniter4/shield/blob/develop/docs/quickstart.md#customize-login-redirect
